lib/3scale/backend/transactor.rb

```diff
@@ -66,29 +66,25 @@ def validate(oauth, provider_key, report_usage, params, request_info)
         # service_id cannot be taken from params since it might be missing there
         service_id = service.id
 
-        app_id = params[:app_id]
-        # TODO: make sure params are nil if they are empty up the call stack
+        # Make sure params are nil if they are empty up the call stack
         # Note: app_key is an exception, as it being empty is semantically
         # significant.
-        params[:app_id] = nil if app_id && app_id.empty?
+        app_id = params[:app_id]&.empty? ? nil : params[:app_id]
+        user_key = params[:user_key]&.empty? ? nil : params[:user_key]
 
         # While OIDC without an app_id makes little sense, we would break existing
         # behaviour when calling non oauth_auth*.xml endpoints if we returned an
         # error here, so only do this for oauth_auth*.xml endpoints.
         raise ApplicationNotFound.new nil if oauth && app_id.nil?
 
-        validators = if oidc_service
-                       Validators::OIDC_VALIDATORS
-                     elsif oauth
-                       Validators::OAUTH_VALIDATORS
-                     else
-                       Validators::VALIDATORS
-                     end
+        raise AuthenticationError if app_id && user_key
+
+        if service.backend_version.to_i == 1
```

Comment:
I think there was a quesiton already, shouldn't we fail the request if version is not `1` or `2`? Now we differenciate between `1` and whatever else.

Reply:
Not in this method, because here `backend_version` can also be `oauth`:

https://github.com/3scale/apisonator/blob/ed5cbe80bd58a0bab41c676e92b0b28bbe4601ba/lib/3scale/backend/transactor.rb#L64

Expected values here are empty, `nil`, `1`, `2` and `oauth`. Look at this comment https://github.com/3scale/apisonator/pull/428#discussion_r2191684480. As I interpreted that test, this means: regarding keys validation, when `backend_version` is `1` we authenticate by `user_key`,  in all other scenarios we authenticate by `app_id + app_key`.